Move CUDA linalg code to its own subfolder

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72306
* __->__ #72304

This is a no-op change that simply moves files around in preparation of moving linear algebra in its own dynamically boundable module
This also simplifies torch_cuda_cu build rules, as all files from linalg it needs are in its own folder now

Differential Revision: [D33992796](https://our.internmc.facebook.com/intern/diff/D33992796)

malfet committed:
Move CUDA linalg code to its own subfolder

This is a no-op change that simply moves files around in preparation of moving linear algebra in its own dynamically boundable module
This also simplifies torch_cuda_cu build rules, as all files from linalg it needs are in its own folder now

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33992796).

IvanYashchuk commented on aten/src/ATen/native/sparse/cuda/SparseCUDABlas.cpp:
```diff
@@ -6,6 +6,7 @@
 #include <c10/cuda/CUDACachingAllocator.h>
 
 #include <cusparse.h>
+#include <ATen/native/cuda/linalg/CUDASolver.h>
```
Why is this include needed?

malfet replied:
See https://github.com/pytorch/pytorch/pull/72306 - it creates `torch_cuda_linalg` as the only library depending on cudasolver and magma, so I move `CUDASolver.[cpp|h] header to `native/cuda/linalg/` folder

IvanYashchuk replied:
This file doesn't seem to use anything from `<ATen/native/cuda/linalg/CUDASolver.h>` because it's cuSPARSE specific, not cuSOLVER.

malfet replied:
Good point, removed.

malfet committed:
Update on "Move CUDA linalg code to its own subfolder"


This is a no-op change that simply moves files around in preparation of moving linear algebra in its own dynamically boundable module
This also simplifies torch_cuda_cu build rules, as all files from linalg it needs are in its own folder now

Differential Revision: [D33992796](https://our.internmc.facebook.com/intern/diff/D33992796)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33992796).

malfet committed:
Update on "Move CUDA linalg code to its own subfolder"


This is a no-op change that simply moves files around in preparation of moving linear algebra in its own dynamically boundable module
This also simplifies torch_cuda_cu build rules, as all files from linalg it needs are in its own folder now

Differential Revision: [D33992796](https://our.internmc.facebook.com/intern/diff/D33992796)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33992796).

malfet committed:
Update on "Move CUDA linalg code to its own subfolder"


This is a no-op change that simply moves files around in preparation of moving linear algebra in its own dynamically boundable module
This also simplifies torch_cuda_cu build rules, as all files from linalg it needs are in its own folder now

Differential Revision: [D33992796](https://our.internmc.facebook.com/intern/diff/D33992796)

[ghstack-poisoned]

malfet committed:
Update on "Move CUDA linalg code to its own subfolder"


This is a no-op change that simply moves files around in preparation of moving linear algebra in its own dynamically boundable module
This also simplifies torch_cuda_cu build rules, as all files from linalg it needs are in its own folder now

Differential Revision: [D33992796](https://our.internmc.facebook.com/intern/diff/D33992796)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33992796).
